c_api/tests/c_api_behavior_test.cpp

```diff
@@ -113,3 +113,453 @@ TEST_F(CapiTest, TestBehaviorGetRuntimeCapabilities) {
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+    }
+  };
+  inner(TI_ARCH_VULKAN);
+}
+
+TEST_F(CapiTest, TestBehaviorMapMemory) {
+  TiMemoryAllocateInfo allocate_info;
+  allocate_info.size = 1024;
+  allocate_info.usage = TI_MEMORY_USAGE_STORAGE_BIT;
+
+  auto inner = [&](TiArch arch) {
+    if (ti::is_arch_available(arch)) {
+      TiRuntime runtime = ti_create_runtime(arch);
+      TiMemory memory = ti_allocate_memory(runtime, &allocate_info);
+      ti_map_memory(runtime, memory);
+      CHECK_TAICHI_SUCCESS();
+
+      // runtime & memory are both null
+      {
+        ti_map_memory(TI_NULL_HANDLE, TI_NULL_HANDLE);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is null, memory is valid
+      {
+        TiRuntime runtime = ti_create_runtime(arch);
+        TiMemory memory = ti_allocate_memory(runtime, &allocate_info);
+        ti_map_memory(TI_NULL_HANDLE, memory);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is valid, memory is null
+      {
+        TiRuntime runtime = ti_create_runtime(arch);
+        ti_map_memory(runtime, TI_NULL_HANDLE);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+    }
+  };
+  inner(TI_ARCH_VULKAN);
+}
+
+TEST_F(CapiTest, TestBehaviorUnmapMemory) {
+  TiMemoryAllocateInfo allocate_info;
+  allocate_info.size = 1024;
+  auto inner = [&](TiArch arch) {
+    if (ti::is_arch_available(arch)) {
+      TiRuntime runtime = ti_create_runtime(arch);
+      for (int i = 0; i < 4; ++i) {
+        allocate_info.usage = TI_MEMORY_USAGE_STORAGE_BIT << i;
+        TiMemory memory = ti_allocate_memory(runtime, &allocate_info);
+        ti_map_memory(runtime, memory);
+        ti_unmap_memory(runtime, memory);
+        CHECK_TAICHI_SUCCESS();
+      }
+
+      // runtime & memory are both null
+      {
+        ti_unmap_memory(TI_NULL_HANDLE, TI_NULL_HANDLE);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is null, memory is valid
+      {
+        TiRuntime runtime = ti_create_runtime(TI_ARCH_VULKAN);
+        TiMemory memory = ti_allocate_memory(runtime, &allocate_info);
+        ti_unmap_memory(TI_NULL_HANDLE, memory);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is valid, memory is valid
+      {
+        TiRuntime runtime = ti_create_runtime(TI_ARCH_VULKAN);
+        ti_unmap_memory(runtime, TI_NULL_HANDLE);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+    }
+  };
+  inner(TI_ARCH_VULKAN);
+}
+
+TEST_F(CapiTest, TestBehaviorAllocateImage) {
+  TiImageExtent extent;
+  extent.height = 512;
+  extent.width = 512;
+  extent.depth = 1;
+  extent.array_layer_count = 1;
+  TiImageAllocateInfo imageAllocateInfo;
+  imageAllocateInfo.dimension = TI_IMAGE_DIMENSION_2D;
+  imageAllocateInfo.format = TI_FORMAT_RGBA8;
+  imageAllocateInfo.extent = extent;
+  imageAllocateInfo.usage = TI_IMAGE_USAGE_STORAGE_BIT;
```

Comment:
Check if it sets a `TI_ERROR_ARGUMENT_OUT_OF_RANGE` if usage is `TI_IMAGE_USAGE_MAX_ENUM`.